packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -274,59 +332,60 @@ export class ProjectLifecycleManager {
+
+  readyToInitialize (projectRoot: string): boolean {
+    this.verifyProjectRoot(projectRoot)
+
     const { needsCypressJsonMigration } = this.refreshMetaState()
 
-    const legacyConfigPatah = path.join(projectRoot, this.legacyConfigFile)
+    const legacyConfigPath = path.join(projectRoot, this.legacyConfigFile)
 
-    if (needsCypressJsonMigration && !this.ctx.isRunMode && this.ctx.fs.existsSync(legacyConfigPatah)) {
-      // we run the legacy plugins/index.js in a child process
-      // and mutate the config based on the return value for migration
-      // only used in open mode (cannot migrate via terminal)
-      const legacyConfig = this.ctx.fs.readJsonSync(legacyConfigPatah) as LegacyCypressConfigJson
+    if (needsCypressJsonMigration && !this.ctx.isRunMode && this.ctx.fs.existsSync(legacyConfigPath)) {
+      this.legacyMigration(legacyConfigPath).catch(this.onLoadError)
 
-      // should never throw, unless there existing pluginsFile errors out,
-      // in which case they are attempting to migrate an already broken project.
-      this.ctx.actions.migration.initialize(legacyConfig)
-      .then(this._pendingMigrationInitialize?.resolve)
-      .finally(() => this._pendingMigrationInitialize = undefined)
-      .catch(this.onLoadError)
+      return false
     }
 
+    this.legacyPluginGuard()
+
     this.configFileWarningCheck()
 
-    if (this.metaState.hasValidConfigFile) {
-      // at this point, there is not a cypress configuration file to initialize
-      // the project will be scaffolded and when the user selects the testing type
-      // the would like to setup
-      this.initializeConfig().catch(this.onLoadError)
-    }
+    return this.metaState.hasValidConfigFile
+  }
 
-    this.loadCypressEnvFile().catch(this.onLoadError)
+  async legacyMigration (legacyConfigPath: string) {
+    try {
+      // we run the legacy plugins/index.js in a child process
+      // and mutate the config based on the return value for migration
+      // only used in open mode (cannot migrate via terminal)
+      const legacyConfig = this.ctx.fs.readJsonSync(legacyConfigPath) as LegacyCypressConfigJson
 
-    if (this.ctx.coreData.currentTestingType) {
-      this.setCurrentTestingType(this.ctx.coreData.currentTestingType)
-    }
+      // should never throw, unless there existing pluginsFile errors out,
+      // in which case they are attempting to migrate an already broken project.
+      await this.ctx.actions.migration.initialize(legacyConfig)
 
-    // If migration is needed only initialize the watchers
-    // when the migration is done.
-    //
-    // NOTE: If we watch the files while initializing,
-    // the config will be loaded before the migration is complete.
-    // The migration screen will disappear see `Main.vue` & `MigrationAction.ts`
-    if (!needsCypressJsonMigration) {
-      this.initializeConfigWatchers()
+      this.ctx.emitter.toLaunchpad()
+    } catch (error) {
+      this.onLoadError(error)
     }
   }
 
```

Comment:
Why not have used a setter function for `setRunModeExitEarly()` ?
`set runModeExitEarly()` would have been fine, right?

Reply:
+1. This was existing code that I copied, but I can update it.

Reply:
Added here: [`6bace98` (#20775)](https://github.com/cypress-io/cypress/pull/20775/commits/6bace98a24172d9dcaa72b133eee180e7edba875)